apps/src/javalab/Playground.js

```diff
@@ -81,17 +98,33 @@ export default class Playground {
   }
 
   addClickableItem(itemData) {
-    if (this.isGameOver) {
-      // can't add new items if the game is over
-      return;
-    }
+    this.addImageHelper(itemData, true);
   }
 
   addImageItem(itemData) {
-    if (this.isGameOver) {
-      // can't add new items if the game is over
+    this.addImageHelper(itemData, false);
+  }
+
+  addImageHelper(itemData, isClickable) {
+    // ignore request if the game is over or if the item already exists
+    if (this.isGameOver || this.imageItemExists(itemData)) {
       return;
     }
+    let onClick = isClickable
+      ? () => this.handleImageClick(itemData.id)
+      : () => {};
+
+    const imageData = {
+      fileUrl: this.getUrl(itemData.filename),
+      x: itemData.x,
+      y: itemData.y,
+      height: itemData.height,
+      width: itemData.width,
+      index: itemData.index,
+      onClick: onClick,
+      type: PlaygroundItemType.IMAGE
```

Comment:
Future thought - this will depend on how text is implemented I think, but I wonder if it might make sense down the road for Javabuilder to send this 'type' as part of the item data, instead of Javalab inferring it from the message? Looks like we're handling text and images mostly in the same way, aside from converting file name -> URL and assigning the appropriate react component, so I'm thinking it could help streamline things (but can definitely be a post launch refactor if anything).

Reply:
I think this could be something to consider once we implement the text side to see how much duplication there is. The message types are different so the type is a 1:1 conversion.